Lesson3.ipynb

```diff
@@ -29,6 +32,440 @@
+   "metadata": {},
+   "source": [
+    "Let's do a practice with `while` loop:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# use while loop to print numbers divisible by 6 between 1 and 40 (each number is on one line)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Function\n",
+    "\n",
+    "### What is a function?\n",
+    "In computer programming, a function (subroutine) is a sequence of program instructions that perform a specific task, packaged as a unit. This unit can then be used in programs wherever that particular task should be performed. (Wikipedia)\n",
+    "\n",
+    "When I attending a Bluejeans meeting, here are what I usually do:\n",
+    "> Open Bluejeans\n",
+    "\n",
+    "> Enter the meeting ID - 4482147579 \n",
+    "\n",
+    "> Participate in the meeting\n",
+    "\n",
+    "> Leave the meeting at the end\n",
+    "\n",
+    "> Enter the rating\n",
+    "\n",
+    "These steps are usually similar for my other Bluejeans meetings, except the 2nd step I might enter a different meeting ID\n",
+    "\n",
+    "In coding we can define a function like this:\n",
+    "```Python\n",
+    "def join_bluejeans_meeting(meeting_id):\n",
+    "    ...                               # omitted to save space\n",
+    "    enter_meeting_id using meeting_id # making use of the meeting_id in parameter\n",
+    "    ...                               # omitted to save space\n",
+    "    \n",
+    "# if I want to attend meeting 4482147579, I can do\n",
+    "join_bluejeans_meeting(4482147579)\n",
+    "\n",
+    "# if I want to join another meeting 5111111111, I can do\n",
+    "join_bluejeans_meeting(5111111111)\n",
+    "```\n",
+    "\n",
+    "Function is basically a group of commands to the computer, and it can come with parameters to modify the behaviour of these commands.\n",
+    "\n",
+    "General syntax to define a function:\n",
+    "```Python\n",
+    "def functionname( parameters ):\n",
+    "    operations inside the function # take note of the indentation, operations and return indented belong to this function\n",
+    "                                   # we can put any kind of command inside the function, e.g. variable assignment, conditional, loop, etc.\n",
+    "    return some_value # if some_value is missing, function returns None. If return statement is missing, function also returns None\n",
+    "```\n",
+    "\n",
+    "Syntax to execute a function:\n",
+    "```Python\n",
+    "functionname( parameters ) # we need substitute parameters with their real values\n",
+    "```\n",
+    "\n",
+    "Here are a few examples:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Function to return sum of 3 numbers\n",
+    "def sum_of_three(num1, num2, num3): # we define the function here\n",
```

Comment:
Do we need to explain coding convention? When to use _ and when to use camelBack?

Reply:
I think it is fine, python convention is _ . If we want convention probably should explain way earlier